src/Quasar/FS/Mount.purs

```diff
@@ -19,35 +19,129 @@ module Quasar.FS.Mount where
 import Prelude
 
 import Data.Argonaut (Json, decodeJson, (.?))
+import Data.Const (Const(..))
 import Data.Either (Either(..))
-import Data.Generic.Rep (class Generic)
-import Data.Generic.Rep.Show (genericShow)
+import Data.Eq (class Eq1, eq1)
+import Data.Identity (Identity(..))
 import Data.Maybe (Maybe)
+import Data.Newtype (unwrap)
 import Data.Path.Pathy (DirName, FileName, dir, file, pathName, (</>))
-import Quasar.Mount.Type as MT
-import Quasar.Types (AnyPath, DirPath)
+import Quasar.Types (AnyPath, DirPath, FilePath)
 
-data Mount = Mount MT.MountType AnyPath
+data MountF f
+  = View (f FilePath)
+  | Module (f DirPath)
+  | MongoDB (f DirPath)
+  | Couchbase (f DirPath)
+  | MarkLogic (f DirPath)
+  | SparkHDFS (f DirPath)
+  | SparkLocal (f DirPath)
+  | Mimir (f DirPath)
+  | Unknown String (f AnyPath)
 
-derive instance eqMount ∷ Eq Mount
-derive instance ordMount ∷ Ord Mount
-derive instance genericMount ∷ Generic Mount _
-instance showMount ∷ Show Mount where show = genericShow
+type Mount = MountF Identity
+type MountType = MountF (Const Unit)
+
+
+instance eqMount ∷ (Eq1 f) => Eq (MountF f) where
+  eq inX inY =  case inX of
+    View x
+      | View y <- inY -> eq1 x y
+      | otherwise -> false
+    Module x
+      | Module y <- inY -> eq1 x y
+      | otherwise -> false
+    MongoDB x
+      | MongoDB y <- inY -> eq1 x y
+      | otherwise -> false
+    Couchbase x
+      | Couchbase y <- inY -> eq1 x y
+      | otherwise -> false
+    MarkLogic x
+      | MarkLogic y <- inY -> eq1 x y
+      | otherwise -> false
+    SparkHDFS x
+      | SparkHDFS y <- inY -> eq1 x y
+      | otherwise -> false
+    SparkLocal x
+      | SparkLocal y <- inY -> eq1 x y
+      | otherwise -> false
+    Mimir x
+      | Mimir y <- inY -> eq1 x y
+      | otherwise -> false
+    Unknown xName x
+      | Unknown yName y <- inY -> eq1 x y && eq xName yName
+      | otherwise -> false
+
+instance showMount ∷ (Show (f String), Functor f) => Show (MountF f) where
```

Comment:
I'd use `TacitString` here otherwise it ends up `show`ing values that have already been `show`n, giving double escaping.